sys/linux/dev_snd_hw.txt

```diff
@@ -0,0 +1,135 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <uapi/asm/ioctl.h>
+include <uapi/linux/fcntl.h>
+include <uapi/linux/time.h>
+include <uapi/sound/asound.h>
+include <uapi/sound/firewire.h>
+
+resource fd_snd_hw[fd]
+
+syz_open_dev$sndhw(dev ptr[in, string["/dev/snd/hwC#D#"]], id intptr, flags flags[open_flags]) fd_snd_hw
+
+read$sndhw(fd fd_snd_hw, buffer ptr[out, snd_firewire_event], count bytesize[buffer])
```

Comment:
It doesn't usually make sense to describe specific structures copied to userspace by the `read` syscall. The only case when this matters, is when the read structure contains a resource that might be reused by other syscalls. Let's just have a generic `read$sndhw` with a generic `ptr[out, array[int8]]` as the argument.

Reply:
So, I shouldn't also keep the structures/ unions needed in those read syscalls then?